flink-runtime/src/main/scala/org/apache/flink/runtime/taskmanager/TaskManager.scala

```diff
@@ -129,6 +129,25 @@ class TaskManager(val connectionInfo: InstanceConnectionInfo,
     override def getValue: Double =
       ManagementFactory.getOperatingSystemMXBean().getSystemLoadAverage()
   })
+  metricRegistry.register("cpuLoad", new Gauge[Double] {
+    override def getValue: Double = {
+      try{
+        val osMXBean = ManagementFactory.getOperatingSystemMXBean().
+          asInstanceOf[com.sun.management.OperatingSystemMXBean]
+        return fetchCPULoad(osMXBean).asInstanceOf[Double]
+      } catch {
```

Comment:
On JVMs not having the `OperatingSystemMXBean` we'll log a full stack trace on each heartbeat (each time we collect metrics).

I would only register the "cpuLoad" metric if we know that we can cast the OSMXBean to `OperatingSystemMXBean` and when the `getProcessCpuLoad()` method is available.
Otherwise, I would register a dummy method that is always returning -1